src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -330,8 +330,8 @@ LedgerMaster::setValidLedger(std::shared_ptr<Ledger const> const& l)
 
     if (!standalone_)
     {
-        auto const vals =
-            app_.getValidations().getTrustedForLedger(l->info().hash);
+        auto vals = app_.getValidations().getTrustedForLedger(l->info().hash);
+        filterValsWithnUnl(vals, app_.validators().getnUnlNodeIDs());
```

Comment:
Samething here, i dont think we should call filterValsWithnUnl if the amendment is not activated. even though I don't think it will do anything. good to have conditional here

Reply:
To check if the amendment is enabled, we need the ledger. This function is called 5 times in the code, we don't always have the ledger, such as this one:
https://github.com/pwang200/rippled/blob/807dd247c678a188af2f609af6edbdbdfc53ac4f/src/ripple/app/ledger/impl/LedgerMaster.cpp#L950

Reply:
I can add comments in code to explain.